### DISCUSSION TITLE
Coinbase SmartWallet

### DISCUSSION THREAD

**Discussion by agsola** [rainbow-me/rainbowkit#1999] [General]
I'm trying to integrate Coinbase Smartwallet with the latest 2.1.x release using dev.
I can create the smart wallet, and then, when trying to Sign in with Ethereum most of the time I'm getting this:
![image](https://github.com/rainbow-me/rainbowkit/assets/472079/7d1979b1-ea8e-4c30-9429-aad1efcaea87)
It seems the network id is always 1 (which could make sense as there is no account connected). However, it leaves us little option to test it.

Are there any instructions to test it?

And, no idea who I got it to work one time, but the Sign in with Ethereum failed despite having support for ERC-4361.

**Comment by 0xmkh** [comment]
@agsola Coinbase smart wallet is only supported on localhost with testnet chains like sepolia. On production you'd have to connect with your coinbase wallet app or coinbase wallet extension. [Here](https://www.rainbowkit.com/docs/wallet-button) is a small example with wallet button. We're also working with supporting siwe for smart wallets [here](https://github.com/rainbow-me/rainbowkit/pull/1997).

**Reply by agsola** [reply]
Maybe I didn't explain properly. 
This was all done on localhost. 

I was not able to select network as when connecting I still don't have the chance to select the network. Thus the issue. After connecting, and before choosing network, the Sign In with Ethereum is prompted. Therefore, the default chain is chosen (1, aka Ethereum). And thus the issue.

**Reply by 0xmkh** [reply]
Got it. At the moment we haven't added siwe support for smart wallets, we're working on that. Regarding the select network not working i think you're not using testnet chains right ? Currently Coinbase smart wallet only supports testnet chains on localhost like sepolia. Can you try that instead ?

**Reply by agsola** [reply]
I was using them, but also other networks. Just leaving the base sepolia network did the trick, thank you!

**Reply by ayushgupta0610** [reply]
Has the support been added now for the smart wallets such as Coinbase smart wallet?

**Reply by DanielSinclair** [reply]
@ayushgupta0610 Yes, Smart Wallet is supported today using the `coinbaseWallet` connector
